Src/Sarif.PartnerMatcher.Function/SpamAnalyzer.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+using Microsoft.CodeAnalysis.Sarif;
+using Microsoft.CodeAnalysis.Sarif.Driver;
+using Microsoft.CodeAnalysis.Sarif.PatternMatcher;
+using Microsoft.CodeAnalysis.Sarif.Writers;
+
+using Newtonsoft.Json;
+
+namespace Microsoft.CodeAnalysis.Sarif.PatternMatcher.Function
+{
+    public class SpamAnalyzer
+    {
+        public static SarifLog Analyze(string filePath, string text, string rulePath)
+        {
+            string logContents = text;
+
+            string regexDefinitions = Path.Combine(rulePath, @"..\Rules\SEC1001.json");
+
+            var fileSystem = new FileSystem();
+
+            // Load all rules from JSON. This also automatically loads any validations file that 
+            // lives alongside the JSON. For a JSON file named PlaintextSecrets.json, the 
+            // corresponding validations assembly is named PlaintextSecrets.dll (i.e., only the
+            // extension name changes from .json to .dll).
+            ISet<Skimmer<AnalyzeContext>> skimmers =
+                AnalyzeCommand.CreateSkimmersFromDefinitionsFiles(fileSystem, new string[] { regexDefinitions });
+
+            var sb = new StringBuilder();
+
+            using (var outputTextWriter = new StringWriter(sb))
+            using (var logger = new SarifLogger(
+                outputTextWriter,
+                LoggingOptions.PrettyPrint | LoggingOptions.Verbose,
+                dataToRemove: OptionallyEmittedData.NondeterministicProperties))
+            {
+                // The analysis will disable skimmers that raise an exception. This 
+                // hash set stores the disabled skimmers. When a skimmer is disabled, 
+                // that catastrophic event is logged as a SARIF notification. 
+                var disabledSkimmers = new HashSet<string>();
+
+                var context = new AnalyzeContext
+                {
+                    TargetUri = new Uri(filePath, UriKind.RelativeOrAbsolute),
+                    FileContents = logContents,
+                    Logger = logger
+                };
+
+                using (context)
+                {
+                    AnalyzeCommand.AnalyzeTargetHelper(context, skimmers, disabledSkimmers);
+                }
+            }
+
+            // Now we'll rewrite the log file in order to convert non-deterministic
+            // absolute URLs to some stable relative reference (built off the source
+            // root.
+
+            SarifLog sarifLog = JsonConvert.DeserializeObject<SarifLog>(sb.ToString());
```

Comment:
>SarifLog sarifLog = JsonConvert.DeserializeObject<SarifLog>(sb.ToString()); [](start = 12, length = 75)

since we are exposing text, I think we could skip the conversion and return the sb.ToString()

Reply:
tried to return raw json string but it looks like this:
"{\r\n  \"$schema\": \"https://schemastore.azurewebsites.net/schemas/json/sarif-2.1.0-rtm.5.json\",\r\n  \"version\": \"2.1.0\",\r\n  \"runs\": [\r\n    {\r\n      \"results\": [\r\n        {\r\n          \"ruleId\": \"SEC1001/AzureDevOpsPersonalAccessToken\",\r\n          \"ruleIndex\": 0,\r\n          \"level\": \"error\",\r\n          \"message\": {\r\n            \"id\": \"Default\",\r\n            \"arguments\": [\r\n              \"VariousPats.txt\",\r\n              \"base64-encoded\",\r\n              \"Azure DevOps personal access token\"\r\n            ]\r\n          },\r\n          \"locations\": [\r\n